### PR TITLE
APPS-484: Correct HTTP Error response for exceptions not handled by s…

### PR DESCRIPTION
**Pull request by ravidbro**
…ervice that probably came from calling a different service.

### CODE REVIEW

File: hammock/common.py [hunk 1]
```diff
@@ -75,9 +76,25 @@ def url_join(*parts):
     return '/'.join(arg.strip('/') for arg in parts if arg)
 
 
+def get_exception_message(exc):
+    if not isinstance(exc, Exception):
+        return None
+    if isinstance(exc, requests.exceptions.HTTPError):
```
Comment (shay-stratoscale) [inline hammock/common.py]: you seem to double check the exception type everywhere you use this method

File: hammock/common.py [hunk 2]
```diff
@@ -75,9 +76,25 @@ def url_join(*parts):
     return '/'.join(arg.strip('/') for arg in parts if arg)
 
 
+def get_exception_message(exc):
```
Comment (shay-stratoscale) [inline hammock/common.py]: doc string are your friends

File: hammock/resource.py [hunk 3]
```diff
@@ -69,7 +81,7 @@ def handle_exception(self, exc, exception_handler, request_uuid, request_method,
             # If exception was not converted yet, we convert it to internal server error.
             if not isinstance(exc, exceptions.HttpError):
                 logging.error("Exception is not http-exception. Translating to InternalServerError")
-                exc = exceptions.InternalServerError(repr(exc))
+                exc = exceptions.InternalServerError()
```
Comment (shay-stratoscale) [inline hammock/resource.py]: empty message?

Comment (ravidbro) [reply hammock/resource.py]: yes, in a case of an INTERNAL error, the user shouldn't get from the API the internal reason, 500 is a bug or a system failure and should be debugged through logs/events

File: hammock/common.py (hunk 2 above)
Comment (ravidbro) [reply hammock/common.py]: not when you have a good method name

File: hammock/common.py (hunk 1 above)
Comment (ravidbro) [reply hammock/common.py]: what do you suggest?

**Comment by ravidbro** [comment]
@shay-stratoscale , I don't think it's changes hammock exceptions "API", every hammck exception that is thrown will remain untouched and will be delivered the same way as it was until now, the change here is when the API call service A and service A call service B and service B raises exception - today service A will always translate it to 500.
If service A throws hammock exception you don't enter my new code.
If service B raises **hammock** exception, the client of service B inside service A will raises **requests** exception, and that what I'm handling.
I prefer to do it here and not in a different utility because we want this behavior will work automatically for all services without changing nothing in current service code (except updating hammock hash)

**Comment by shohamp** [comment]
@ravidbro I think this broke symp-cli:
```
[root@stratonode1 ~]# symp
Starting new HTTPS connection (1): virtual-nb.service.strato
__str__ returned non-string (type Error)
[root@stratonode1 ~]# 
```

**Comment by ravidbro** [comment]
symp -k ?

**Pull request merged**